protos/table.proto

```diff
@@ -137,6 +139,9 @@ message Manifest {
   //
   // This specifies what format is used to store the data files.
   DataStorageFormat data_format = 15;
+  
+  // Table metadata.
```

Comment:
Let's state that keys starting with `lance.` are reserved for use by the Lance library. 

```suggestion
  // Table metadata.
  //
  // Keys with the prefix "lance." are reserved for the Lance library. Other
  // libraries may wish to similarly prefix their configuration keys
  // appropriately.
```